pyccel/parser/semantic.py

```diff
@@ -2386,13 +2386,18 @@ def _visit_FunctionDef(self, expr, **settings):
         headers = []
 
         not_used = [d for d in decorators if d not in def_decorators.__all__]
-
+        not_used.remove('templates') #templates is not a decorator
```

Comment:
I'm not sure if this is quit clear, as we already have `template` in decorators which contains a list of `FunctionCall` that each represents a call to the `template` decorator, and then we have `templates` which basically has the same thing, represented in a different manner, it's a bit ambiguous.

Reply:
for me i was thinking about why we should insert a non-decorator variable into a decorators dict, and you told that's to make the code more structured. @EmilyBourne What do you think ?

Reply:
I agree with Redouane. I'm not sure what you mean by "inserting a non-decorator variable into a decorators dict". `@template` is a decorator

Reply:
The new `templates` dict which non-decorator